src/server/pps/server/api_server.go

```diff
@@ -1493,6 +1493,24 @@ func (a *apiServer) validateEnterpriseChecks(ctx context.Context, req *pps.Creat
 	return nil
 }
 
+// validateEgress validates the egress field. We only validate secret for now,
+// because the egress field will be taken out of PPS spec eventually.
+func (a *apiServer) validateEgress(pipelineName string, egress *pps.Egress) error {
+	if egress == nil {
+		return nil
+	}
+	if target, ok := egress.Target.(*pps.Egress_SqlDatabase); ok {
+		secret := target.SqlDatabase.GetSecret()
+		if secret == nil {
+			return errors.New("egress.sql_database.secret is required")
+		}
+		if secret.K8SSecret == "" || secret.Key == "" {
+			return errors.New("egress.sql_database.secret.k8s_secret and egress.sql_database.secret.key are required")
+		}
```

Comment:
I guess also check `Columns` isn't set for `CSV`

Reply:
I actually think `Columns` can be useful for both CSV and JSON. For example, a user could potentially have data that represents a subset of the columns in the target table, and want all the other columns to be filled with null values.